lib/simulator_sse.h

```diff
@@ -0,0 +1,599 @@
+// Copyright 2019 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SIMULATOR_SSE_H_
```

Comment:
As I understand it, there was a previous attempt to add SSE to qsim - were any changes made between that version and this one?

+@MichaelBroughton for additional insight on this.

Reply:
Yes, there was a previous attempt to add SSE to qsim. It closely followed the AVX version using the same memory layout. The current SSE version uses a different memory layout, which is more suitable for SSE. This version is also much shorter than the previous one.